### PR TITLE
add the s3 configuration of encrypt access key and secret key

### PR DESCRIPTION
**Pull request by tigershi**

### CODE REVIEW

**Review by linr211 — APPROVED** [review]

**Review by Xiaochao8 — APPROVED** [review]

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java [hunk 1]
```diff
@@ -6,67 +6,116 @@
...
-    */
-   @Value("${s3.bucketName}")
-   private String bucketName;
+	public String getAccessKey() {
+		if (this.encryption) {
+			try {
+				return RsaCryptUtils.decryptData(this.accessKey, this.publicKey);
+			} catch (Exception e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java:22]: Remove the comment and log the error message?

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java [hunk 2]
```diff
@@ -6,67 +6,116 @@
...
+		this.accessKey = accessKey;
+	}
+
+	public String getSecretkey() {
+		if (this.encryption) {
+			try {
+				return RsaCryptUtils.decryptData(this.secretkey, this.publicKey);
+			} catch (Exception e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java:40]: same as above

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java [hunk 3]
```diff
@@ -6,67 +6,116 @@
...
+	public String getPublicKey() {
+		return publicKey;
+	}
+
+	public void setPublicKey(String publicKey) {
+		this.publicKey = publicKey;
+	}
+
+	/**
+	 * the s3 password is encryption or not
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java:85]:
```suggestion
	 * the s3 password is encrypted or not
```

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java [hunk 4]
```diff
@@ -6,67 +6,116 @@
...
+	 */
+	@Value("${s3.password.encryption:false}")
+	private Boolean encryption;
+	
+
+	/**
+	 * the s3 password public key use to decrypt data
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/S3Config.java:92]:
```suggestion
	 * the public key use to decrypt data
```

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/RsaCryptUtils.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+/**
+ * 
+ * it use to decode the access key and Secret key
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/RsaCryptUtils.java:22]:
```suggestion
 * This class is to decode the access key and Secret key
```

File: g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/RsaCryptUtils.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ */
+public class RsaCryptUtils {
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/conf/RsaCryptUtils.java:25]: Please check this issue reported by SONAR:
https://sonarcloud.io/organizations/singleton/rules?open=java%3AS5542&rule_key=java%3AS5542

**Pull request merged**